integration/base/src/main/java/com/inrupt/client/integration/base/ApplicationRequestMetadataScenarios.java

```diff
@@ -0,0 +1,298 @@
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.inrupt.client.integration.base;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import com.inrupt.client.Headers;
+import com.inrupt.client.Request;
+import com.inrupt.client.Response;
+import com.inrupt.client.auth.Credential;
+import com.inrupt.client.auth.Session;
+import com.inrupt.client.openid.OpenIdException;
+import com.inrupt.client.openid.OpenIdSession;
+import com.inrupt.client.solid.SolidClient;
+import com.inrupt.client.solid.SolidClientException;
+import com.inrupt.client.solid.SolidRDFSource;
+import com.inrupt.client.solid.SolidSyncClient;
+import com.inrupt.client.spi.RDFFactory;
+import com.inrupt.client.util.URIBuilder;
+import com.inrupt.client.webid.WebIdProfile;
+
+import java.net.URI;
+import java.util.*;
+import java.util.stream.Stream;
+
+import org.apache.commons.rdf.api.Dataset;
+import org.apache.commons.rdf.api.IRI;
+import org.apache.commons.rdf.api.Literal;
+import org.apache.commons.rdf.api.RDF;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ApplicationRequestMetadataScenarios {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationRequestMetadataScenarios.class);
+    private static String podUrl;
+    private static String webidUrl;
+
+    private static final Config config = ConfigProvider.getConfig();
+
+    private static String issuer;
+
+    private static final RDF rdf = RDFFactory.getInstance();
+
+    //feature is deactivated by default
+    private static final String REQUEST_METADATA_FEATURE = config
+            .getOptionalValue("inrupt.test.request-metadata.feature", String.class)
+            .orElse("false");
+
+    private static final String[] REQUEST_METADATA_HEADERS_THAT_PROPAGATE = config
+            .getValue("inrupt.test.request-metadata-headers-that-propagate", String[].class);
```

Comment:
I would have expected this to be optional. What is the value when the feature isn't being tested?

Reply:
Please make these config values optional. Otherwise, every developer who wants to build the code will need to add those configs to their environment. Plus, the build will fail in CD